index.js

```diff
@@ -31,24 +33,98 @@ exports.register = (server, options) => {
   // expose sentry client at server.plugins['hapi-sentry'].client
   server.expose('client', Sentry);
 
-  server.ext({
-    type: 'onRequest',
-    method(request, h) {
-      if (opts.useDomainPerRequest) {
-        // Sentry looks for current hub in active domain
-        // Therefore simply by creating&entering domain Sentry will create
-        // request scoped hub for breadcrumps and other scope metadata
-        request.__sentryDomain = domain.create();
-        request.__sentryDomain.enter();
-      }
+  // Set up request interceptor for creating and destroying a domain on each request
+  // It'll wrap the request handler returned by the _dispatch function in HAPI core
+  // allowing it to create the domain before any of HAPIs processing starts
+  // thus allowing us to use the normal HAPI extensions to add context to Sentry scopes
+  // https://github.com/hapijs/hapi/blob/c95985e225fa09c4b640a887ccb4be46dbe265bc/lib/core.js#L507-L538
+  function interceptor(next, req, res, ...args) {
```

Comment:
This is pretty complicated. What do you think about calling this function `hapiDispatchInterceptor` to make this more specific? (not sure if that even is accurate, i never wrapped a function like this)

Reply:
It wraps both the Hapi `_dispatch` function, and also intercepts the events from the `http.Server` listener. Prefer not to rename this here, but after the Typescript conversion is merged, otherwise I worry we'll have a lot problems merging that in.

Reply:
ok. i noted that in your last PR